src/problems/MetalAdvectionProblem/test_sne.cpp

```diff
@@ -0,0 +1,550 @@
+AMREX_GPU_MANAGED Real sigma2 = 10. * 11. * kmps;
+AMREX_GPU_MANAGED Real rho01 = 0.0268988 * Const_mH;
+AMREX_GPU_MANAGED Real rho02 = 1.e-5 * 0.0268988 * Const_mH;
+;
+#if 0 // workaround AMDGPU compiler bug
+};                       // namespace
+#endif
+
+struct NewProblem {
+};
+
+template <> struct HydroSystem_Traits<NewProblem> {
+	static constexpr double gamma = 5. / 3.;
+	static constexpr bool reconstruct_eint = true; // Set to true - temperature
+};
+
+template <> struct quokka::EOS_Traits<NewProblem> {
+	static constexpr double gamma = 5. / 3.;
+	static constexpr double mean_molecular_weight = C::m_u;
+	static constexpr double boltzmann_constant = C::k_B;
+};
+
+template <> struct Physics_Traits<NewProblem> {
+	static constexpr bool is_hydro_enabled = true;
+	static constexpr bool is_radiation_enabled = false;
+	static constexpr bool is_chemistry_enabled = false;
+	static constexpr bool is_mhd_enabled = false;
+	static constexpr int numMassScalars = 0;    // number of mass scalars
+	static constexpr int numPassiveScalars = 1; // number of passive scalars
+	static constexpr int nGroups = 1;	    // number of radiation groups
+};
+
+template <> struct SimulationData<NewProblem> {
+
+	std::unique_ptr<amrex::TableData<Real, 1>> blast_x;
+	std::unique_ptr<amrex::TableData<Real, 1>> blast_y;
+	std::unique_ptr<amrex::TableData<Real, 1>> blast_z;
+
+	int nblast = 0;
+	int SN_counter_cumulative = 0;
+	Real SN_rate_per_vol = NAN;  // rate per unit time per unit volume
+	Real E_blast = 1.0e51;	     // ergs
+	Real M_ejecta = 5.0 * Msun;  // 5.0 * Msun; // g
+	Real refine_threshold = 1.0; // gradient refinement threshold
+};
+
+template <> void QuokkaSimulation<NewProblem>::setInitialConditionsOnGrid(quokka::grid const &grid_elem)
+{
+
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> dx = grid_elem.dx_;
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = grid_elem.prob_lo_;
+	const amrex::Box &indexRange = grid_elem.indexRange_;
+	const amrex::Array4<double> &state_cc = grid_elem.array_;
+
+	double vol = AMREX_D_TERM(dx[0], *dx[1], *dx[2]);
+
+	amrex::ParallelFor(indexRange, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+		amrex::Real const z = prob_lo[2] + (k + amrex::Real(0.5)) * dx[2];
+
+		// Calculate DM Potential
+		double prefac;
+		prefac = 2. * M_PI * Const_G * rho_dm * std::pow(R0, 2);
+		double Phidm = (prefac * std::log(1. + std::pow(z / R0, 2)));
+
+		// Calculate Stellar Disk Potential
+		double prefac2;
+		prefac2 = 2. * M_PI * Const_G * Sigma_star * z_star;
+		double Phist = prefac2 * (std::pow(1. + z * z / z_star / z_star, 0.5) - 1.);
+
+		// Calculate Gas Disk Potential
+
+		double Phigas;
+		// Interpolate to find the accurate g-value from array-- because linterp doesn't work on Setonix
+		// TODO - AV to find out why linterp doesn't work
+		size_t ii = 0;
+		double x_interp = std::abs(z);
+		amrex::GpuArray<amrex::Real, 64> xx = z_data;
+		amrex::GpuArray<amrex::Real, 64> yy = logphi_data;
```

Comment:
@aditivijayan @markkrumholz My guess why the compiler fails on Setonix is that it's trying to allocate registers for each array element here. You should be able to access `z_data` and `logphi_data` directly without copying them into local arrays.

Reply:
I am guessing this because the compiler fails during register allocation for this function:
```
2.      Running pass 'Greedy Register Allocator' on function '@_ZN5amrex13launch_globalILi256EZNS_11ParallelForILi256EZN18RadhydroSimulationI10NewProblemE26setInitialConditionsOnGridEN6quokka4gridEEUliiiE_EENSt9enable_ifIXsr19MaybeDeviceRunnableIT0_EE5valueEvE4typeERKNS_3Gpu10KernelInfoERKNS_3BoxERKS9_EUlvE_EEvS9_.intern.9c1f5acb36cd098c'
```